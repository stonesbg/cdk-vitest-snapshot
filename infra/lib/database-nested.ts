import * as cdk from "aws-cdk-lib";
import * as dynamodb from "aws-cdk-lib/aws-dynamodb";

import { Construct } from "constructs";

interface DatabaseResourcesProps extends cdk.NestedStackProps {}

export class DatabaseResources extends cdk.NestedStack {
  public table: dynamodb.Table;

  constructor(scope: Construct, id: string, props: DatabaseResourcesProps) {
    super(scope, id, props);

    // our database is a static resource
    this.table = new dynamodb.Table(this, "test", {
      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
      partitionKey: {
        name: "id",
        type: dynamodb.AttributeType.STRING,
      },
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });
  }
}
