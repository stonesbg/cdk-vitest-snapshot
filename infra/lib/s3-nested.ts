import * as cdk from "aws-cdk-lib";
import * as s3 from "aws-cdk-lib/aws-s3";

import { Construct } from "constructs";

interface S3ResourcesProps extends cdk.NestedStackProps {}

export class S3Resources extends cdk.NestedStack {
  constructor(scope: Construct, id: string, props: S3ResourcesProps) {
    super(scope, id, props);

    // S3 Bucket
    new s3.Bucket(this, "TestBucket", {
      bucketName: "my-test-bucket",
      versioned: true,
    });
  }
}
