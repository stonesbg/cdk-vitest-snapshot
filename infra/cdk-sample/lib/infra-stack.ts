import * as cdk from "aws-cdk-lib";
import { Construct } from "constructs";
import * as lambda from "aws-cdk-lib/aws-lambda";
import * as apigw from "aws-cdk-lib/aws-apigateway";
import * as s3 from "aws-cdk-lib/aws-s3";

import { DatabaseResources } from "./database-nested";
import { S3Resources } from "./s3-nested";

export class InfraStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // S3 Bucket
    new s3.Bucket(this, "TestBucket", {
      bucketName: "my-test-bucket",
      versioned: true,
    });

    const fn = new lambda.Function(this, "MyFunction", {
      code: lambda.Code.fromAsset("../../apps/lambda-handler"),
      runtime: lambda.Runtime.NODEJS_LATEST,
      handler: "index.handler",
    });

    const fn2 = new lambda.Function(this, "from-bucket", {
      code: lambda.Code.fromBucket(
        cdk.aws_s3.Bucket.fromBucketName(this, "MyBucket", "saved-assets"),
        "assets/lambda-handler.zip",
      ),
      runtime: lambda.Runtime.NODEJS_LATEST,
      handler: "index.handler",
    });

    const endpoint = new apigw.LambdaRestApi(this, "MyEndpoint", {
      handler: fn,
      restApiName: "HelloApi",
    });

    // we pull in the two stateful nested stacks
    new DatabaseResources(this, "DatabaseResources", {});

    new S3Resources(this, "EventBusResources", {});
  }
}
