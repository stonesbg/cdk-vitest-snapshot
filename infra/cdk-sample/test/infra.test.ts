import { test, expect } from "vitest";
import * as cdk from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import * as CdkAppSample from "../lib/infra-stack";

test("CDK Sample App Stack", () => {
  const app = new cdk.App();
  const stack = new CdkAppSample.InfraStack(app, "CdkAppSampleStack");
  const template = Template.fromStack(stack);

  expect(template).toMatchSnapshot();
});
