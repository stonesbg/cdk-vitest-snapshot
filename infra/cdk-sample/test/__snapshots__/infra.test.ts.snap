// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CDK Sample App Stack 1`] = `
{
  "Outputs": {
    "MyEndpoint20B13C5B": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "https://",
            {
              "Ref": "MyEndpointE262FD4F",
            },
            ".execute-api.",
            {
              "Ref": "AWS::Region",
            },
            ".",
            {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            {
              "Ref": "MyEndpointDeploymentStageprodCD409E34",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "DatabaseResourcesNestedStackDatabaseResourcesNestedStackResourceAC95A73F": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "[ASSET BUCKET]",
              },
              "/[ASSET TEMPLATEURL]",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "EventBusResourcesNestedStackEventBusResourcesNestedStackResourceA61EE6C6": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "[ASSET BUCKET]",
              },
              "/[ASSET TEMPLATEURL]",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
    "MyEndpointANY0BA96E02": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "MyFunction3BAA72D1",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Fn::GetAtt": [
            "MyEndpointE262FD4F",
            "RootResourceId",
          ],
        },
        "RestApiId": {
          "Ref": "MyEndpointE262FD4F",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "MyEndpointANYApiPermissionCdkAppSampleStackMyEndpoint34D5FFBFANY029C285C": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "MyFunction3BAA72D1",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "MyEndpointE262FD4F",
              },
              "/",
              {
                "Ref": "MyEndpointDeploymentStageprodCD409E34",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "MyEndpointANYApiPermissionTestCdkAppSampleStackMyEndpoint34D5FFBFANYB80E4DC7": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "MyFunction3BAA72D1",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "MyEndpointE262FD4F",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "MyEndpointAccountA97F5175": {
      "DeletionPolicy": "Retain",
      "DependsOn": [
        "MyEndpointE262FD4F",
      ],
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "MyEndpointCloudWatchRole45929D3D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
      "UpdateReplacePolicy": "Retain",
    },
    "MyEndpointCloudWatchRole45929D3D": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
      "UpdateReplacePolicy": "Retain",
    },
    "MyEndpointDeploymentBAE6EF2C222837e5b5d6be87eed40314bcaad0bf": {
      "DependsOn": [
        "MyEndpointproxyANYA11EF0E3",
        "MyEndpointproxy67989C2F",
        "MyEndpointANY0BA96E02",
      ],
      "Properties": {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": {
          "Ref": "MyEndpointE262FD4F",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "MyEndpointDeploymentStageprodCD409E34": {
      "DependsOn": [
        "MyEndpointAccountA97F5175",
      ],
      "Properties": {
        "DeploymentId": {
          "Ref": "MyEndpointDeploymentBAE6EF2C222837e5b5d6be87eed40314bcaad0bf",
        },
        "RestApiId": {
          "Ref": "MyEndpointE262FD4F",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "MyEndpointE262FD4F": {
      "Properties": {
        "Name": "HelloApi",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "MyEndpointproxy67989C2F": {
      "Properties": {
        "ParentId": {
          "Fn::GetAtt": [
            "MyEndpointE262FD4F",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": {
          "Ref": "MyEndpointE262FD4F",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "MyEndpointproxyANYA11EF0E3": {
      "Properties": {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                {
                  "Fn::GetAtt": [
                    "MyFunction3BAA72D1",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": {
          "Ref": "MyEndpointproxy67989C2F",
        },
        "RestApiId": {
          "Ref": "MyEndpointE262FD4F",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "MyEndpointproxyANYApiPermissionCdkAppSampleStackMyEndpoint34D5FFBFANYproxy4DD39086": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "MyFunction3BAA72D1",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "MyEndpointE262FD4F",
              },
              "/",
              {
                "Ref": "MyEndpointDeploymentStageprodCD409E34",
              },
              "/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "MyEndpointproxyANYApiPermissionTestCdkAppSampleStackMyEndpoint34D5FFBFANYproxyD777E5CC": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "MyFunction3BAA72D1",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Ref": "MyEndpointE262FD4F",
              },
              "/test-invoke-stage/*/*",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "MyFunction3BAA72D1": {
      "DependsOn": [
        "MyFunctionServiceRole3C357FF2",
      ],
      "Properties": {
        "Code": "[CODE REMOVED]",
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "MyFunctionServiceRole3C357FF2",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "MyFunctionServiceRole3C357FF2": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "TestBucket560B80BC": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketName": "my-test-bucket",
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "frombucket42B2B247": {
      "DependsOn": [
        "frombucketServiceRole4D28DBC6",
      ],
      "Properties": {
        "Code": "[CODE REMOVED]",
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "frombucketServiceRole4D28DBC6",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "frombucketServiceRole4D28DBC6": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
